README.md
# react-dashboard-agenda

> Dashboard Agenda for react

[![NPM](https://img.shields.io/npm/v/react-dashboard-agenda.svg)](https://www.npmjs.com/package/react-dashboard-agenda) [![JavaScript Style Guide](https://img.shields.io/badge/code_style-standard-brightgreen.svg)](https://standardjs.com)

## Install

```bash
npm install --save @yana4961/react-dashboard-agenda
```

## React version
React >17.0.2

## Usage

import ChunksArea and stylesheet

```jsx
import React from 'react'

import { ChunksArea } from 'react-dashboard-agenda'
import 'react-dashboard-agenda/dist/index.css'

const App = () => {
  const handleClickChunk = (from, to, state, more) => {
    console.log('click on', from, ' - ', to, ', isAllowTime: ', state, more)
  }

  return (
    <div style={{ padding: '10px 20px' }}>
      <ChunksArea
        id='my_dashboard'
        chunksTime={60}
        chunksStyle={{
          margin: '4px 4px'
        }}
        date='2021-08-27'
        classesChunks={['fixedSize']}
        chunksFree={[
          { from: '2021-08-27T06:00:00+10:00', to: '2021-08-27T18:00:00+10:00'},
          { from: '2021-08-27T19:00:00+10:00', to: '2021-08-27T20:00:00+10:00'}
        ]}
        locale='Asia/Vladivostok'
        handleClickChunk={handleClickChunk}
        reserveTimes={[
          {
            id: 1,
            from: '2021-08-27T08:00:00+10:00',
            to: '2021-08-27T09:00:00+10:00',
            type: 'meetup',
            data: {
              title: 'Планерка'
            }
          },
          {
            id: 2,
            from: '2021-08-27T12:00:00+10:00',
            to: '2021-08-27T14:00:00+10:00',
            type: 'social',
            data: {
              title: 'Встреча с клиентами'
            }
          },
          {
            id: 3,
            from: '2021-08-27T12:00:00+10:00',
            to: '2021-08-27T15:00:00+10:00',
            type: 'meetup',
            data: {
              title: 'Вечеринка'
            }
          }
        ]}
        types={[{id: 'meetup', color: '#E9F6FB'}, { id: 'social', color: '#f7c0ea' }]}
      />
    </div>

  )
}

export default App
```

## License

MIT © [YanaLugina &lt;yana4961@gmail.com&gt;](https://github.com/YanaLugina &lt;yana4961@gmail.com&gt;)


CHANGELOG.md
## 2021-11-05
Was added memo on chunks and chunkArea, fixed ids and keys for chunks

## 2021-09-28
Was added labels on chunks with time from and time to for chunk. Was created around avatar for xLegenda.

## 2021-09-24
 Was created XLegend component for x-legend information

## 2021-09-22 
Was created x-legend for chunks, added data for x-legends, sorted reserves for resources legend

## 2021-09-14
Was created legend for chunks. Was renamed classes for classesChunks

## 2021-09-13 
Was created timesChunks container for x legend,
was added props for create custom style chunks

## 2021-09-09
Was created filter for picked layers

## 2021-09-07
Was added new props with checked time ranges and with data for set under agenda, was added prop with layers colors and ids.

## 2021-08-31
Was added util for get free times from array chunksFree with objects and was marked denied chunks

## 2021-08-27
Was added function from date-fns for create time from/to for chunks,
was added id for component, was created argument for set locale, was added function for tap on chunk.

## Create chunks default 2021-08-25

## Create project 2021-08-24


example/src/App.js
import React from 'react'

import { ChunksArea } from 'react-dashboard-agenda'
import 'react-dashboard-agenda/dist/index.css'
import img1 from './assets/avatar~hgQOkIra.svg'
import img2 from './assets/avatar2~IUSznSPb.svg'

const App = () => {
  const handleClickChunk = (from, to, state) => {
    console.log('click on', from, ' - ', to, 'dataFromChunk: ', state)
  }

  return (
    <div style={{ padding: '10px 20px' }}>
      <ChunksArea
        id='my_dashboard'
        chunksTime={60}
        chunksStyle={{marginLeft: 4, marginRight: 4}}
        date='2021-08-27'
        classesChunks={['fixedSize']}
        chunksFree={[
          { from: '2021-08-27T06:00:00+10:00', to: '2021-08-27T18:00:00+10:00'},
          { from: '2021-08-27T19:00:00+10:00', to: '2021-08-27T20:00:00+10:00'}
        ]}
        locale='Asia/Vladivostok'
        handleClickChunk={handleClickChunk}
        reserveTimes={[
          {
            id: 1,
            from: '2021-08-27T08:00:00+10:00',
            to: '2021-08-27T09:00:00+10:00',
            type: 'meetup',
            resourceId: 1,
            data: {
              title: 'Планерка'
            }
          },
          {
            id: 2,
            from: '2021-08-27T12:00:00+10:00',
            to: '2021-08-27T14:00:00+10:00',
            type: 'social',
            resourceId: 1,
            data: {
              title: 'Встреча с клиентами'
            }
          },
          {
            id: 3,
            from: '2021-08-27T10:00:00+10:00',
            to: '2021-08-27T11:00:00+10:00',
            type: 'social',
            resourceId: 2,
            data: {
              title: 'Встреча с клиентами'
            }
          },
          {
            id: 4,
            from: '2021-08-27T12:00:00+10:00',
            to: '2021-08-27T15:00:00+10:00',
            type: 'meetup',
            resourceId: 2,
            data: {
              title: 'Вечеринка'
            }
          },
          {
            id: 4,
            from: '2021-08-27T12:00:00+10:00',
            to: '2021-08-27T14:00:00+10:00',
            type: 'meetup',
            resourceId: 1,
            data: {
              title: 'Вечеринка'
            }
          }
        ]}
        types={[{id: 'meetup', color: 'rgb(233, 246, 251)'}, { id: 'social', color: 'rgb(247, 192, 234)' }]}
        resources={[
          { id: 1, label: 'first', title: 'First hero', description: 'La-la-la', image: img1 },
          { id: 2, label: 'second', title: 'Second hero', description: 'Sa-sa-sa', image: img2 }
        ]}
      />
    </div>

  )
}

export default App
